src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticListener.cs

```diff
@@ -196,7 +231,8 @@ public override void Write(string name, object value)
         private class DiagnosticSubscription : IDisposable
         {
             internal IObserver<KeyValuePair<string, object>> Observer;
-            internal Predicate<string> IsEnabled;
+
```

Comment:
I may be over-optimizing, but I feel bad that we are making the COMMON IsEnabled case slower for the sake of the rare new case (by forcing all calls through the new IsEnabled).   I would like to avoid this by keeping a field for both the 1 arg and 3 arg case.   When you subscribe you set BOTH fields (This if the subscriber gives the 1 arg IsEnabled, you fix up the 3 arg field to call the 1 arg function, and do the reverse if the subscribe gives you the 3 arg case.  

This insures that if there is no mismatch between supplier and subscriber, the call is efficient (matches).